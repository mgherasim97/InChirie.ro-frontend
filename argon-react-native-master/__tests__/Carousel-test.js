
/**
 * @jest-environment jsdom
 */

import React from 'react';
import Carousel from 'react-native-anchor-carousel';
import { mount, shallow } from 'enzyme';
import {
  StyleSheet,
  Dimensions,
} from 'react-native';

// import jsdom from 'jsdom'
// const doc = jsdom.jsdom('<!doctype html><html><body></body></html>')
// global.document = doc
// global.window = doc.defaultView

var jsdom = require('jsdom');
const { JSDOM } = jsdom;

const { document } = (new JSDOM('')).window;
global.document = document;

// const React = require('react')
// const Inregistrare = require('../screens/Inregistrare')
// const renderer = require('react-test-renderer')

const styles = StyleSheet.create({
  carousel: {
    flex: 1,
    backgroundColor: '#141518'
  }
});


describe('Image Carousel rendering', () => {

  it('Renders without crashing', () => {

    const component = shallow(<Carousel />);
  });

  
  it('Renders correct data', () => {
    const component = shallow(<Carousel />);

    expect(component).toMatchSnapshot();
  });

  it('The expected data is shown', () => {
    const component = mount(<Carousel />);

    component.setProps({data: [{"content": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...", "title": "Lorem ipsum dolor sit amet", "uri": "https://i.imgur.com/GImvG4q.jpg"}]});
    expect(component.props().data).toEqual([{"content": "Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...", "title": "Lorem ipsum dolor sit amet", "uri": "https://i.imgur.com/GImvG4q.jpg"}]);
   

    component.unmount();

  });

  it('The given style is used', () => {
    const component = mount(<Carousel />);

    component.setProps({style: styles.carousel});
    expect(component.props().style).toEqual(styles.carousel);

    component.unmount();

  });

  it('The correct item width is used', () => {
    const component = mount(<Carousel />);
    
    const { width } = Dimensions.get('window');

    component.setProps({itemWidth: 3 * width});
    expect(component.props().itemWidth).toEqual(3 * width);

    component.unmount();

  });

  it('The correct container width is used', () => {
    const component = mount(<Carousel />);
    
    const { width } = Dimensions.get('window');

    component.setProps({containerWidth: width});
    expect(component.props().containerWidth).toEqual(width);

    component.unmount();

  });
  
  it('The correct container inActiveOpacity is used', () => {
    const component = mount(<Carousel />);
    
    const { width } = Dimensions.get('window');

    component.setProps({inActiveOpacity: 3});
    expect(component.props().inActiveOpacity).toEqual(3);

    component.unmount();

  });


});

