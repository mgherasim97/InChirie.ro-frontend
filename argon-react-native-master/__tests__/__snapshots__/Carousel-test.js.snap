// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Carousel rendering Renders correct data 1`] = `
<AnimatedComponent
  automaticallyAdjustContentInsets={false}
  bounces={true}
  data={Array []}
  decelerationRate={0}
  getItemLayout={[Function]}
  horizontal={true}
  inActiveOpacity={0.8}
  inActiveScale={0.8}
  initialScrollIndex={0}
  itemContainerStyle={Object {}}
  keyExtractor={[Function]}
  minScrollDistance={25}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEnd={[Function]}
  onScrollEndDrag={[Function]}
  pagingEnable={true}
  renderItem={[Function]}
  separatorWidth={0}
  showsHorizontalScrollIndicator={false}
  style={
    Array [
      Object {},
      Object {
        "width": 750,
      },
      Object {},
    ]
  }
/>
`;
